crates/bws/src/command/secret.rs

```diff
@@ -0,0 +1,166 @@
+    output_settings: OutputSettings,
+) -> Result<()> {
+    let secret = client
+        .secrets()
+        .get(&SecretGetRequest { id: secret_id })
+        .await?;
+    serialize_response(secret, output_settings.output, output_settings.color);
+
+    Ok(())
+}
+
+pub(crate) async fn create(
+    mut client: Client,
+    organization_id: Uuid,
+    secret: SecretCreateCommandModel,
+    output_settings: OutputSettings,
+) -> Result<()> {
+    let secret = client
+        .secrets()
+        .create(&SecretCreateRequest {
+            organization_id,
+            key: secret.key,
+            value: secret.value,
+            note: secret.note.unwrap_or_default(),
+            project_ids: Some(vec![secret.project_id]),
+        })
+        .await?;
+    serialize_response(secret, output_settings.output, output_settings.color);
+
+    Ok(())
+}
+
+pub(crate) async fn edit(
+    mut client: Client,
+    organization_id: Uuid,
+    secret: SecretEditCommandModel,
+    output_settings: OutputSettings,
+) -> Result<()> {
+    let old_secret = client
+        .secrets()
+        .get(&SecretGetRequest { id: secret.id })
+        .await?;
+
+    let new_secret = client
+        .secrets()
+        .update(&SecretPutRequest {
+            id: secret.id,
+            organization_id,
+            key: secret.key.unwrap_or(old_secret.key),
+            value: secret.value.unwrap_or(old_secret.value),
+            note: secret.note.unwrap_or(old_secret.note),
+            project_ids: match secret.project_id {
+                Some(id) => Some(vec![id]),
+                None => match old_secret.project_id {
+                    Some(id) => Some(vec![id]),
+                    None => bail!("Editing a secret requires a project_id."),
+                },
+            },
+        })
+        .await?;
+    serialize_response(new_secret, output_settings.output, output_settings.color);
+
+    Ok(())
+}
+
+pub(crate) async fn delete(mut client: Client, secret_ids: Vec<Uuid>) -> Result<()> {
+    let count = secret_ids.len();
+
+    let result = client
+        .secrets()
+        .delete(SecretsDeleteRequest { ids: secret_ids })
+        .await?;
+
+    let secrets_failed: Vec<(Uuid, String)> = result
+        .data
+        .into_iter()
+        .filter_map(|r| r.error.map(|e| (r.id, e)))
+        .collect();
+    let deleted_secrets = count - secrets_failed.len();
+
+    match deleted_secrets {
+        2.. => println!("{} secrets deleted successfully.", deleted_secrets),
+        1 => println!("{} secret deleted successfully.", deleted_secrets),
+        _ => (),
+    }
+
+    match secrets_failed.len() {
+        2.. => eprintln!("{} secrets had errors:", secrets_failed.len()),
+        1 => eprintln!("{} secret had an error:", secrets_failed.len()),
+        _ => (),
+    }
+
+    for secret in &secrets_failed {
+        eprintln!("{}: {}", secret.0, secret.1);
+    }
+
+    if !secrets_failed.is_empty() {
+        process::exit(1);
```

Comment:
I think it would be cleaner to return an Err here and have the process exit handling in `main.rs`.

Reply:
Updated in https://github.com/bitwarden/sdk/commit/f599033914f1e8740cac3cd64f6f1c0927915500.